optaplanner-examples/src/main/java/org/optaplanner/examples/cheaptime/solver/drools/IdleCost.java

```diff
@@ -16,12 +16,21 @@
 
 package org.optaplanner.examples.cheaptime.solver.drools;
 
-import org.apache.commons.lang3.builder.CompareToBuilder;
+import java.util.Comparator;
+
 import org.apache.commons.lang3.builder.EqualsBuilder;
 import org.apache.commons.lang3.builder.HashCodeBuilder;
 import org.optaplanner.examples.cheaptime.domain.Machine;
 
-public class IdleCost {
+import static java.util.Comparator.comparing;
+import static java.util.Comparator.comparingLong;
+
+public class IdleCost implements Comparable<IdleCost> {
+
+    private static final Comparator<IdleCost> COMPARATOR =
+            comparing(IdleCost::getMachine, comparingLong(Machine::getId))
```

Comment:
The original code did:
  .append(machine, other.machine)
Why not just compare the machines, as AbstractPersistable still implements Comparable?

Reply:
@ge0ffrey `AbstractPersistable` does not implement `Comparable`. That's the point of this PR.

Reply:
Ah yes of course. Ignore this comment.